src/utils/PostRequestComponent.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+
+interface ApiResponse {
+  status: number;
+  json: () => Promise<{ message?: string; error?: string }>;
+}
+
+const mockApiCall = (): Promise<ApiResponse> => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const success = Math.random() > 0.2;
+      if (success) {
+        resolve({
+          status: 200,
+          json: () => Promise.resolve({ message: "Success!" }),
+        });
+      } else {
+        reject({
+          status: 500,
+          json: () => Promise.resolve({ error: "Server Error" }),
+        });
+      }
```

Comment:
issue: the mockApiCall does not fully mimic a real Fetch API Response. Right now, it returns a custom object instead of a proper Response. This may cause issues when integrating with fetch() or other request-handling functions. 

**Suggested Fix: Modify mockApiCall to return a real Response-like object:**
```
      const body = success
        ? { message: "Success!" }
        : { error: "Server Error" };

      const response = new Response(JSON.stringify(body), {
        status: success ? 200 : 500,
        headers: { "Content-Type": "application/json" },
      });

      if (success) {
        resolve(response);
      } else {
        reject(response);
      }
```